crates/astria-conductor/src/executor/queue.rs

```diff
@@ -0,0 +1,504 @@
+        } else {
+            None
+        }
+    }
+
+    fn pop_head_blocks(&mut self) -> Option<Vec<SequencerBlockSubset>> {
+        if let Some(head_blocks) = self.pending_blocks.get_mut(&self.head_height) {
+            let mut output_blocks: Vec<SequencerBlockSubset> = vec![];
+            let tmp_blocks = head_blocks.clone();
+            let mut blocks: Vec<&SequencerBlockSubset> = tmp_blocks.values().collect();
+            blocks.sort();
+            for block in blocks {
+                output_blocks.push(block.clone());
+            }
+            let most_recent_height = output_blocks[output_blocks.len() - 1].height();
+            self.head_height = most_recent_height.increment();
+            self.remove_data_blow_height(self.head_height);
+            return Some(output_blocks);
+        }
+        None
+    }
+
+    // check to see if the block is already present in the queue
+    fn is_block_present(&mut self, block: &SequencerBlockSubset) -> bool {
+        let block_hash = block.block_hash();
+        let height = block.height();
+
+        // check if the block is already present in the pending blocks
+        if let Some(pending_blocks) = self.pending_blocks.get(&height) {
+            if let Some(_block) = pending_blocks.get(&block_hash) {
+                return true;
+            }
+        }
+        // check if the block is already present in the soft blocks
+        if let Some(soft_block) = self.soft_blocks.get(&height) {
+            if soft_block.block_hash() == block_hash {
+                return true;
+            }
+        }
+
+        false
+    }
+
+    fn is_block_a_parent(&mut self, block: SequencerBlockSubset) -> bool {
+        let block_hash = block.block_hash();
+        if let Some(child_blocks) = self.pending_blocks.get(&block.child_height()) {
+            let blocks = child_blocks.values();
+            for block in blocks {
+                if let Some(hash) = block.parent_hash() {
+                    if hash == block_hash {
+                        return true;
+                    }
+                }
+            }
+        }
+        false
+    }
+
+    fn insert_to_pending_blocks(&mut self, block: SequencerBlockSubset) {
+        let height = block.height();
+        let block_hash = block.block_hash();
+
+        if let Some(pending_blocks) = self.pending_blocks.get_mut(&height) {
+            let mut new_map = pending_blocks.clone();
+            new_map.insert(block_hash, block);
+            self.pending_blocks.insert(height, new_map);
+        } else {
+            let mut new_map = HashMap::new();
+            new_map.insert(block_hash, block);
+            self.pending_blocks.insert(height, new_map);
+        }
+    }
+
+    // remove all data in the queue below a given height. this does not remove
+    // data from the soft queue, only the pending queue, and updates the head height.
+    // TODO: add error handling
+    fn remove_data_blow_height(&mut self, height: Height) {
```

Comment:
```suggestion
    fn remove_data_below_height(&mut self, height: Height) {
```

Not sure if the `blow` was intentional